src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -11745,6 +11745,105 @@ def str_len(self, **kwargs: Any) -> "SnowflakeQueryCompiler":
             self._modin_frame.apply_snowpark_function_to_data_columns(length)
         )
 
+    def str_slice(
```

Comment:
we should reuse loc or iloc indexing for this.

Reply:
Are those applicable here? loc/iloc are extracting a subset of values from a column, but Series.str.slice is applying a substring operation to every element in the column.

Reply:
Exactly, what @sfc-gh-joshi said. We'd still need to worry about the same space of edge cases though!